hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/streaming/HoodieStreamSource.scala

```diff
@@ -71,19 +69,6 @@ class HoodieStreamSource(
     parameters.get(DataSourceReadOptions.QUERY_TYPE.key).contains(DataSourceReadOptions.QUERY_TYPE_INCREMENTAL_OPT_VAL) &&
     parameters.get(DataSourceReadOptions.INCREMENTAL_FORMAT.key).contains(DataSourceReadOptions.INCREMENTAL_FORMAT_CDC_VAL)
 
-  /**
-   * When hollow commits are found while doing streaming read , unlike batch incremental query,
-   * we do not use [[HollowCommitHandling.FAIL]] by default, instead we use [[HollowCommitHandling.BLOCK]]
-   * to block processing data from going beyond the hollow commits to avoid unintentional skip.
-   *
-   * Users can set [[DataSourceReadOptions.INCREMENTAL_READ_HANDLE_HOLLOW_COMMIT]] to
-   * [[HollowCommitHandling.USE_TRANSITION_TIME]] to avoid the blocking behavior.
-   */
-  private val hollowCommitHandling: HollowCommitHandling =
```

Comment:
Should we remove the class `HollowCommitHandling` because it is not used now.

Reply:
We still need to handle hollow commits if the source table is version 6 or below.  So keeping this for now and let HUDI-8359 fix it properly.

Reply:
> We still need to handle hollow commits if the source table is version 6 or below

Are you saying we fallback to to instant time incremental queries? Maybe just telling user the semantics is migrated for 1.0 release reader.

Reply:
We discussed this issue and agreed that it's simpler to deprecate the hollow commit handling in Hudi 1.0 as the completion time-based incremental query can properly support serializability; user should upgrade the table to Hudi 1.0 for the correct behavior.

Regarding this PR, there are many places that need clean-up on the hollow commit handling.  I'll defer this to a separate PR.